--- ./jscomp/others/dune
+++ ./jscomp/others/dune
@@ -54,7 +54,7 @@
   (progn
    (with-stdout-to
     %{target}
-    (run awk -f %{dep:expand_module_aliases_node.awk} %{deps})))))
+    (bash "awk -f %{dep:expand_module_aliases_node.awk} %{deps}")))))
 
 (library
  (modes melange)

--- ./jscomp/stdlib/dune
+++ ./jscomp/stdlib/dune
@@ -5,7 +5,7 @@
   (progn
    (with-stdout-to
     %{target}.awk
-    (run awk -f %{dep:expand_module_aliases.awk} %{deps}))
+    (bash "awk -f %{dep:expand_module_aliases.awk} %{deps}"))
    (run cppo -D=BS %{env:CPPO_FLAGS=} %{target}.awk -o %{target}.awk)
    (with-stdout-to
     %{target}
@@ -20,7 +20,7 @@
   (progn
    (with-stdout-to
     %{target}.awki
-    (run awk -f %{dep:expand_module_aliases.awk} %{deps}))
+    (bash "awk -f %{dep:expand_module_aliases.awk} %{deps}"))
    (run cppo -D=BS %{env:CPPO_FLAGS=} %{target}.awki -o %{target}.awki)
    (with-stdout-to
     %{target}

